File: src/main/java/com/zerozero/repository/UserRepository.java
package com.zerozero.repository;

import com.zerozero.domain.User;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {

  Optional<User> findByEmail(String email);
  boolean existsByNickname(String nickname);
  boolean existsByEmail(String email);
}

File: src/main/java/com/zerozero/user/domain/repository/UserRepository.kt
package com.zerozero.user.domain.repository

import com.zerozero.user.domain.model.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, UUID> {

    fun findByEmail(email: String): Optional<User>

    fun findAllByIdIn(userIds: List<UUID>): List<User>
}
